rules.py

```diff
@@ -911,6 +911,22 @@ def check(self):
             raise base.ElectionTreeError(
                 "The Election File has following issues with the identifiers.", error_log)
 
+class CandidatesMissingPartyData(base.TreeRule):
```

Comment:
I'd actually make this a base.BaseRule instead of a TreeRule. It'll make the errors that get printed much more readable. See the `OptionalAndEmpty` rule for a template on how to do that.